Scripts/RemediationScripts/Remediate-DontUseNonADIdentities.ps1

```diff
@@ -352,22 +385,27 @@ function Remove-AzTSNonADIdentities
         Write-Host "Path: $($folderpath)NonADAccountsRoleAssignments.json"
     }
 
-    if(-not $Force)
-    {
-        Write-Host "Do you want to delete the above listed role assignment? " -ForegroundColor Yellow -NoNewline
-        $UserInput = Read-Host -Prompt "(Y|N)"
+    # Safe Check: Exporting all role assignments in CSV file.
+    $externalAccountsRoleAssignments | Export-CSV -Path "$($folderpath)\NonADAccountsRoleAssignments.csv" -NoTypeInformation
 
-        if($UserInput -ne "Y")
+    if(-not $DryRun)
+    {
+        if(-not $Force)
         {
-            return;
+            Write-Host "Do you want to delete the above listed role assignment? " -ForegroundColor Yellow -NoNewline
+            $UserInput = Read-Host -Prompt "(Y|N)"
+
+            if($UserInput -ne "Y")
+            {
+                return;
+            }
         }
-    }
    
 
-    Write-Host "Step 3 of 3: Clean up Non-AD identities for Subscription [$($SubscriptionId)]..."
+        Write-Host "Step 3 of 3: Clean up Non-AD identities for Subscription [$($SubscriptionId)]..."
     
-    # Start deletion of all Non-AD identities.
-    Write-Host "Starting to delete role assignments for Non-AD identities..." -ForegroundColor Cyan
+        # Start deletion of all Non-AD identities.
+        Write-Host "Starting to delete role assignments for Non-AD identities..." -ForegroundColor Cyan
```

Comment:
Please fix the indentation in lines 410 and 411.

Reply:
Indentation corrected.